conda_lock/src_parser/__init__.py

```diff
@@ -59,32 +73,66 @@ class URLDependency(_BaseDependency):
 Dependency = Union[VersionedDependency, URLDependency]
 
 
+class SourceDependency(StrictModel):
+    dep: Dependency
+    selectors: Selectors = Selectors()
```

Comment:
Can you explain in words what's the idea behind a SourceDependency?

Reply:
Sure (I can also add a comment). So we want the LockSpecification to be a mapping from environment (platform right now, but potentially other attributes in the future) to a list of dependencies. Those dependencies don't need selectors, because selectors are only used when constructing the LockSpecification for determining if a dep is required for an environment.

Thus, a SourceDependency represents a dependency with any additional info associated with it from a source file. Right now, its just selectors, but in the future, we may have other limiters like min or max Python version.

Reply:
Would something like this make sense as a docstring?

> A `SourceDependency` represents information about a particular dependency specification which has been extracted from a `SourceFile` (e.g. `environment.yml` or `pyproject.toml`). Generally, information about the target environment (e.g. platform) is not included, but it might be specified by including selectors.

Please rewrite in case I misunderstood the details.

Reply:
That's perfect! Will add